### PR TITLE
Add warning message to add allele specific annotation group

### PR DESCRIPTION
**Pull request by ronlevine**
Implements https://github.com/broadinstitute/gatk/issues/2983.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java [hunk 1]
```diff
@@ -368,8 +372,9 @@ private static void removeStaleAttributesAfterMerge(final Map<String, Object> at
                     values.add(Integer.parseInt(stringValue));
                 }
             } catch (final NumberFormatException e) {
-                // nothing to do
-                //TODO This comes directly from gatk3, we should determine if we should log this or if it happens so much it is not useful to log
+                final String baseMsg = "Remaining (non-reducible) annotations are assumed to be ints or doubles, but " + value + " doesn't parse and will not be annotated in the final VC.";
+                final String msg = value.toString().contains("|") ? baseMsg + " Add -G Standard -G AS_Standard to the command to annotate in the final VC." : baseMsg;
```
Comment (ronlevine) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java]: From @droazen This seems like a pretty brittle way of detecting whether AS annotations are present...

**Comment by droazen** [comment]
@lbergelson please review

**Comment by jamesemery** [comment]
@lbergelson Could you take a quick look at this one? I just updated it to be in line with the changes I've made to our annotaiton framework.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java [hunk 2]
```diff
@@ -382,6 +385,7 @@ private void addReferenceConfidenceAttributes(final VCWithNewAlleles vcPair, fin
                     values.add(parseNumber(value.toString()));
                 } catch (final NumberFormatException e) {
                     warning.warn(String.format("Detected invalid annotations: When trying to merge variant contexts at location %s:%d the annotation %s was not a numerical value and was ignored",vcPair.getVc().getContig(),vcPair.getVc().getStart(),p.toString()));
+                    if (key.startsWith("AS")) AS_warning.warn(String.format("Reducible annotation '%s' detected, add -G Standard -G AS_Standard to the command to annotate in the final VC with this annotation.",key));
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java]: I always like `{}` on if statements even for 1 liners...

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java [hunk 3]
```diff
@@ -255,4 +255,14 @@ public void testIntervalsAndOnlyOutputCallsStartingInIntervalsAreMutuallyRequire
         args.addArgument("L", "20:69512-69513");
         runCommandLine(args);
     }
+
+    @Test
+    public void testMissingAlleleSpecificAnnotationGroup() throws IOException {
```
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/GenotypeGVCFsIntegrationTest.java]: this doesn't really test anything

**Comment by droazen** [comment]
@jamesemery Can you remove the non-functional test, and merge this once tests pass?

**Pull request merged**